src/components/asset-row.tsx

```diff
@@ -1,42 +1,49 @@
 import React from 'react';
-import { Flex, Text, Box, BoxProps } from '@stacks/ui';
+import { Box, Stack, StackProps } from '@stacks/ui';
+import { Text, Caption } from '@components/typography';
 import { AssetAvatar } from '@components/stx-avatar';
 
-interface AssetRowProps extends BoxProps {
+interface AssetRowProps extends StackProps {
   name: string;
+  contractAddress?: string;
   friendlyName: string;
   value: string;
   subtitle: string;
 }
+
 export const AssetRow = React.forwardRef<HTMLDivElement, AssetRowProps>((props, ref) => {
-  const { name, friendlyName, value, subtitle, ...otherProps } = props;
+  const { name, contractAddress, friendlyName, value, subtitle, ...rest } = props;
   return (
-    <Box width="100%" mb="base" {...otherProps} ref={ref}>
-      <Flex flexWrap="wrap" flexDirection="row" cursor="pointer">
-        <Box width="32px" py="tight" mr="base">
-          <AssetAvatar
-            useStx={name === 'STX' || name === 'Stacks Token'}
-            gradientString={name}
-            mr="tight"
-            size="32px"
-          >
-            {friendlyName[0]}
-          </AssetAvatar>
-        </Box>
-        <Box flexGrow={1}>
-          <Text display="block" fontSize={2} fontWeight="400" color="ink.1000">
-            {friendlyName}
-          </Text>
-          <Text fontSize={1} color="ink.400">
-            {subtitle}
-          </Text>
-        </Box>
-        <Box textAlign="right" pt="tight">
-          <Text fontWeight="400" color="ink.1000">
-            {value}
-          </Text>
-        </Box>
-      </Flex>
-    </Box>
+    <Stack
+      spacing="base"
+      isInline
+      alignItems="center"
+      flexWrap="wrap"
+      flexDirection="row"
+      cursor="pointer"
+      width="100%"
+      {...rest}
+      ref={ref}
+    >
+      <AssetAvatar
+        useStx={name === 'STX' || name === 'Stacks Token'}
```

Comment:
useStx seems like a strange prop name?

Reply:
Yeah - this is likely a result of code I wrote, but it also seems smelly to have two possible `name`s for a STX asset.

Reply:
I agree, this was there already and I think we should punt on fixing it. This issue is not the goal of the PR. When the UI is updated, it's likely this component will be completely refactored.